src/App.jsx

```diff
@@ -14,45 +16,115 @@ export const goodsFromServer = [
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+export const App = () => {
+  const [sortedGoods, setSortedGoods] = useState([...goodsFromServer]);
+  const [isAlphabetical, setIsAlphabetical] = useState(false);
+  const [isByLength, setIsByLength] = useState(false);
+  const [isReversed, setIsReversed] = useState(false);
+
+  const sortAlphabetically = () => {
+    const newGoods = sortedGoods.sort();
+
+    setSortedGoods(isReversed ? newGoods.reverse() : newGoods);
+    setIsAlphabetical(true);
+    setIsByLength(false);
+  };
+
+  const sortByLength = () => {
+    const newGoods = [...goodsFromServer].sort((a, b) => a.length - b.length);
+
+    setSortedGoods(isReversed ? newGoods.reverse() : newGoods);
+    setIsAlphabetical(false);
+    setIsByLength(true);
+  };
+
+  const reverseGoods = () => {
+    const reversedGoods = sortedGoods.reverse();
+
+    setSortedGoods(reversedGoods);
+    setIsReversed(!isReversed);
+  };
+
+  const resetSorting = () => {
+    setSortedGoods([...goodsFromServer]);
+    setIsAlphabetical(false);
+    setIsByLength(false);
+    setIsReversed(false);
+  };
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  const shouldShowResetButton = !(
+    JSON.stringify(sortedGoods) === JSON.stringify(goodsFromServer)
+  );
+
+  const resetButtonClass = classNames(
+    'button',
+    'is-danger', {
+      'is-light': isAlphabetical
+      || isByLength
+      || isReversed,
+    },
+  );
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={classNames(
+            'button',
+            'is-info', {
+              'is-light': !isAlphabetical,
+            },
+          )}
+          onClick={sortAlphabetically}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={classNames(
+            'button',
+            'is-success', {
+              'is-light': !isByLength,
+            },
+          )}
+          onClick={sortByLength}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={classNames(
+            'button',
+            'is-warning', {
+              'is-light': !isReversed,
+            },
+          )}
+          onClick={reverseGoods}
+        >
+          Reverse
+        </button>
```

Comment:
In future, when you gonna use typescript, don't forget that you can iterate through 'enum' of sort type options, and render these buttons in easier way